ci.py

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python3
+from pathlib import Path
+
+import argparse
+import configparser
+import logging
+import os
+import sys
+import pprint
+from subprocess import check_output, check_call, call, CalledProcessError
+
+import yaml
+import git
+
+logging.basicConfig(stream=sys.stdout, format='[%(levelname)s] %(asctime)s: %(message)s', level=logging.INFO)
+
+DEFAULT_BRANCH = 'main' # we can rewrite it in config
+YAML_CONFIG = 'ci-default.yml'
+YAML_CONFIG_CUSTOM = 'ci.yml'
+SUBMODULES_CONFIG = '.gitmodules'
+GIT_SOURCES_FILE = '.git-sources'
+
+
+class Builder():
+    rootdir = check_output(["git", "rev-parse", "--show-toplevel"]).decode('utf-8').strip()
+
+    def __init__(self):
+        self.global_branch_name = None
+        self.config = {}
+        self.custom_config = {}
+        self.read_custom_config()
+        self.read_config()
+        self.deps = self.config['deps']
+
+    def read_custom_config(self):
+        with open(YAML_CONFIG_CUSTOM, 'r') as f:
+            self.custom_config = yaml.load(f, Loader=yaml.FullLoader)
+
+    def write_custom_config(self, config):
+        with open(YAML_CONFIG_CUSTOM, 'w') as f:
+            yaml.dump(config, f, sort_keys=False)
+
+
+    def read_config(self):
```

Comment:
let's make this read_config return config value and assign self.config in constructor instead of updating self.config inside. it will be easier to read.

Reply:
the same for `read_custom_config`